src/App.js

```diff
@@ -1,39 +1,46 @@
-import React, {useState, useEffect} from 'react';
+import {useState, useEffect} from 'react';
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
-import InputWithLabel from "./InputWithLabel";
+//import InputWithLabel from "./InputWithLabel";
 
-// function useSemiPersistentState() {
-//   const savedTodoList = JSON.parse(localStorage.getItem("savedTodoList")) || [];
-//   const [todoList, setTodoList] = useState(savedTodoList);
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  
+  const fetchData = async() => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+    };
 
-//   useEffect(() => {
-//     localStorage.setItem ("savedTodoList", JSON.stringify(todoList));
-//   }, [todoList]);
-//     return [todoList,setTodoList];
-// }
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
 
+    try {
+      const response = await fetch (url, options);
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+          throw new Error(message);
+      }
+      const data = await response.json();
+     
+      const todos = data.records.map((record) => ({
+        title: record.fields.title,
+        id: record.id,      
+      }));
 
-function App() {
-   const [todoList, setTodoList] = useState([]);
-   const [isLoading, setIsLoading] = useState(true);
-   const [workInProgress, serWorkInProgress] = useState(true);
-
+      setTodoList(todos);
+      setIsLoading(false);
 
-   //async with Promise, resolve, setTime when refresh page in 2 sec  the saved item appears in the list
-   useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
-          },
-        });
-      }, 2000);
-    }).then((result) => {
-      setTodoList(result.data.todoList);
+    } catch (error) {
+      console.log("Error fetching data: ",error);
       setIsLoading(false);
-    });
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
   }, []);
```

Comment:
Nitpick:
This useEffect is missing `fetchData` as a dependency, however when defining functions within a component (App) body, the functions as dependencies must be wrapped in a [useCallback](https://react.dev/reference/react/useCallback) to avoid re-renders. Therefore, I'd recommend that you...

either define the `fetchData` within the useEffect body and invoke it:
```js
useEffect(() => {
  const fetchData = async () => { ...etc }
  fetchData();
}, []);
```
OR, wrap `fetchData` with a `useCallback` function and add it a dependency:
```js
import {useCallback, useState, useEffect} from 'react';

function App() { 
  ...etc

  const fetchData = useCallback(async () => { ...etc }, []);

  useEffect(() => {
    fetchData();
  }, [fetchData]); // add fetchData as a dependency here

  return (...etc)
}
```

Reply:
Thank you Matt. Your comments are always helpful. I was trying to do this step, and the way how I finish was the only working. But your nitpicks made more sense for me. Now I got it . Thank you so much!